microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/compatibility/ExpiredJwtTest.java

```diff
@@ -0,0 +1,79 @@
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JsonWebToken;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JwtClaims;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.JwtHelper;
+import org.jboss.eap.qe.microprofile.jwt.auth.tool.RsaKeyTool;
+import org.jboss.eap.qe.microprofile.jwt.testapp.Endpoints;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.JaxRsTestApplication;
+import org.jboss.eap.qe.microprofile.jwt.testapp.jaxrs.SecuredJaxRsEndpoint;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunAsClient
+@RunWith(Arquillian.class)
+public class ExpiredJwtTest {
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, ExpiredJwtTest.class.getSimpleName() + ".war")
+                .addClass(SecuredJaxRsEndpoint.class)
+                .addClass(JaxRsTestApplication.class)
+                .addAsManifestResource(ExpiredJwtTest.class.getClassLoader().getResource("mp-config-basic.properties"),
+                        "microprofile-config.properties")
+                .addAsManifestResource(ExpiredJwtTest.class.getClassLoader().getResource("pki/key.public.pem"),
+                        "key.public.pem");
+    }
+
+    /**
+     * @tpTestDetails Supply an expired ({@code exp} set in past) JWT to server
+     * @tpPassCrit Expired JWT is rejected and user receives 401/Unauthorized.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    public void supplyExpiredJwtToServerTest(@ArquillianResource URL url) throws URISyntaxException {
+        final URL privateKeyUrl = ExpiredJwtTest.class.getClassLoader().getResource("pki/key.private.pkcs8.pem");
+        if (privateKeyUrl == null) {
+            throw new IllegalStateException("Private key wasn't found in resources!");
+        }
+
+        final RsaKeyTool keyTool = RsaKeyTool.newKeyTool(privateKeyUrl.toURI());
+
+        final String subject = "FAKE_USER";
+        final Instant now = Instant.now();
+        final Instant sooner = now.minus(1, ChronoUnit.HOURS); //note the minus here setting time in past
+
+        final JsonWebToken expiredJwt = JwtHelper.generateProperSignedJwtWithClaims(keyTool, new JwtClaims.Builder()
+                .jwtId(UUID.randomUUID().toString())
+                .audience("microprofile-jwt-testsuite")
+                .subject(subject)
+                .userPrincipalName(subject)
+                .issuer("issuer")
+                .issuedAtTime(now.getEpochSecond())
+                .expirationTime(sooner.getEpochSecond())
+                .groups(Collections.emptySet())
+                .customClaim("preferred_username", subject)
+                .build());
```

Comment:
I am wondering whether the literals that you're using to build the claims here should be defined as constants somewhere, as I see they're actually used in `jwt-auth-tool` module, mainly. WDYT?

Reply:
Thanks for catching this. The custom claim setter can be removed. There is however issue with issuer which should be the same value for each test, so I will think on how to implement this as a constant...

Reply:
Solving in #171  ...

Reply:
Solved.